### PR TITLE
 make sure CFLAGS and MYCFLAGS are used in Lua easyblock

### PR DESCRIPTION
**Pull request by smoors**
needed for ARGoS, to avoid compilation errors like this:
```
/apps/brussel/CO7/ivybridge-ib/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: /apps/brussel/CO7/ivybridge-ib/software/Lua/5.3.5-GCCcore-9.3.0/lib/liblua.a(lapi.o): requires dynamic R_X86_64_32 reloc against 'luaO_nilobject_' which may overflow at runtime; recompile with -fPIC
/apps/brussel/CO7/ivybridge-ib/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: /apps/brussel/CO7/ivybridge-ib/software/Lua/5.3.5-GCCcore-9.3.0/lib/liblua.a(ldebug.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/apps/brussel/CO7/ivybridge-ib/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: /apps/brussel/CO7/ivybridge-ib/software/Lua/5.3.5-GCCcore-9.3.0/lib/liblua.a(ldo.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
/apps/brussel/CO7/ivybridge-ib/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: /apps/brussel/CO7/ivybridge-ib/software/Lua/5.3.5-GCCcore-9.3.0/lib/liblua.a(ldump.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/apps/brussel/CO7/ivybridge-ib/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: /apps/brussel/CO7/ivybridge-ib/software/Lua/5.3.5-GCCcore-9.3.0/lib/liblua.a(lgc.o): requires unsupported dynamic reloc 11; recompile with -fPIC
/apps/brussel/CO7/ivybridge-ib/software/binutils/2.34-GCCcore-9.3.0/bin/ld.gold: error: /apps/brussel/CO7/ivybridge-ib/software/Lua/5.3.5-GCCcore-9.3.0/lib/liblua.a(lmem.o): requires dynamic R_X86_64_32 reloc which may overflow at runtime; recompile with -fPIC
```
also added EB defined `CFLAGS` and optionally user defined `MYCFLAGS`.
tested with Lua-5.3.4 and Lua-5.3.5.

### CODE REVIEW

**Comment by akesandgren** [comment]
That would normally be done by adding pic: true in the toolchainopts

**Comment by smoors** [comment]
> That would normally be done by adding pic: true in the toolchainopts

@akesandgren that's also fine to me. I was not sure where to add it because several other easyblocks also add it unconditionally.

**Comment by smoors** [comment]
this PR is still necessary because the toolchainopts are not picked up without adding CFLAGS.

File: easybuild/easyblocks/l/lua.py [hunk 1]
```diff
@@ -68,6 +68,9 @@ def configure_step(self, cmd_prefix=''):
             mycflags.append('-DLUA_COMPAT_5_1')
         if LooseVersion(self.version) > LooseVersion('5.2'):
             mycflags.append('-DLUA_COMPAT_5_2')
+        if LooseVersion(self.version) > LooseVersion('5.3'):
+            mycflags.append(os.getenv('CFLAGS', ''))
```
Comment (wpoely86) [inline easybuild/easyblocks/l/lua.py:72]: why is this behind an if? It should be used in general?

Comment (boegel) [reply easybuild/easyblocks/l/lua.py:72]: +1

Comment (boegel) [reply easybuild/easyblocks/l/lua.py]: fixed in dda269b and re-tested with all existing Lua easyconfigs, works like a charm

**Pull request merged**